ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/actions/WorkspaceSnapshotCreator.java

```diff
@@ -0,0 +1,88 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2015 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.ide.actions;
+
+import org.eclipse.che.api.promises.client.Operation;
+import org.eclipse.che.api.promises.client.OperationException;
+import org.eclipse.che.api.promises.client.PromiseError;
+import org.eclipse.che.api.workspace.gwt.client.WorkspaceServiceClient;
+import org.eclipse.che.ide.api.notification.Notification;
+import org.eclipse.che.ide.api.notification.NotificationManager;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+/**
+ * Creates snapshot of the workspace using {@link WorkspaceServiceClient}.
+ *
+ * <p>This component is for managing notifications which are related to creating snapshot process.
+ *
+ * @author Yevhenii Voevodin
+ */
+@Singleton
+public class WorkspaceSnapshotCreator {
+
+    private final WorkspaceServiceClient workspaceService;
+    private final NotificationManager    notificationManager;
+
+    Notification notification;
+
+    @Inject
+    public WorkspaceSnapshotCreator(WorkspaceServiceClient workspaceService, NotificationManager notificationManager) {
+        this.workspaceService = workspaceService;
+        this.notificationManager = notificationManager;
+    }
+
+    /**
+     * Changes notification state to finished with an error.
+     */
+    public void creationError(String message) {
+        notification.setMessage(message);
+        notification.setType(Notification.Type.ERROR);
+        notification.setStatus(Notification.Status.FINISHED);
```

Comment:
I think It would be better to have status FAILED. And instead of setting ERROR Type and FINISHED Status we should set only FAILED status. It will be clear and comfortably. But It is another story :)